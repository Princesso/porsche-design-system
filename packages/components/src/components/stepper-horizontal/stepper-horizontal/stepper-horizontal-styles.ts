import { buildResponsiveStyles, getCss } from '../../../utils';
import type { StepperHorizontalSize } from './stepper-horizontal-utils';
import type { BreakpointCustomizable } from '../../../types';
import { fontSizeText, textSmallStyle } from '@porsche-design-system/utilities-v2';

export const getComponentCss = (size: BreakpointCustomizable<StepperHorizontalSize>): string => {
  return getCss({
    '@global': {
      ':host': {
        display: 'block',
      },
    },
    scroller: {
      display: 'flex',
      ...textSmallStyle,
      ...buildResponsiveStyles(size, (s: StepperHorizontalSize) => {
        return { fontSize: fontSizeText[s] };
      }),
    },
  });
};
