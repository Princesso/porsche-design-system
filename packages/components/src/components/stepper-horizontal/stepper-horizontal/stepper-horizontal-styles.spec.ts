import { getComponentCss } from './stepper-horizontal-styles';

xdescribe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    ['small'],
    ['medium'],
    [{ base: 'small', xs: 'medium', s: 'small', m: 'medium', l: 'small', xl: 'medium' }],
  ])('should return correct css for size: %j', (...args) => {
    expect(getComponentCss(...args)).toMatchSnapshot();
  });
});
