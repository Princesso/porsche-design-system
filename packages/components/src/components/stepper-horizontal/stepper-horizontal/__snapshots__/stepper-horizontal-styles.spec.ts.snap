// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for size: "medium" 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
.scroller {
  display: flex;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  font-size: clamp(1.2rem, .12vw + 1.18rem, 1.31rem);
}"
`;

exports[`getComponentCss() should return correct css for size: "small" 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
.scroller {
  display: flex;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  font-size: 1rem;
}"
`;

exports[`getComponentCss() should return correct css for size: {"base":"small","xs":"medium","s":"small","m":"medium","l":"small","xl":"medium"} 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
.scroller {
  display: flex;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  font-size: 1rem;
}
@media(min-width:480px) {
  .scroller {
    font-size: clamp(1.2rem, .12vw + 1.18rem, 1.31rem);
  }
}

@media(min-width:760px) {
  .scroller {
    font-size: 1rem;
  }
}

@media(min-width:1000px) {
  .scroller {
    font-size: clamp(1.2rem, .12vw + 1.18rem, 1.31rem);
  }
}

@media(min-width:1300px) {
  .scroller {
    font-size: 1rem;
  }
}

@media(min-width:1760px) {
  .scroller {
    font-size: clamp(1.2rem, .12vw + 1.18rem, 1.31rem);
  }
}
"
`;
